EventListener/Login.php

```diff
@@ -63,6 +67,28 @@ public function setCustomerId($value)
         $this->options['customerId'] = $value;
     }
 
+    /**
+     * Return an ID or the Username if no ID available.
+     *
+     * @param TokenInterface $token
+     *
+     * @return mixed
+     */
+    protected function getUserIdentifier(TokenInterface $token)
+    {
+        $tokenUser = $token->getUser();
+
+        if ($tokenUser instanceof eZUser || !$tokenUser instanceof UserInterface) {
+            return $tokenUser->getAPIUser()->id;
+        }
+
+        if (method_exists($tokenUser, 'getId')) {
+            return $tokenUser->getId();
+        }
+
+        return $tokenUser->getUsername();
```

Comment:
@clash82 @lserwatka Actually we don't care what ID is used for logged in users, STRINGs are suported here. If the identifier comes from eZ itself, an OAUTH provider or OpenID ... It just needs to be consistent with every login and in sync with the userid being used for getting recommendations after a successful login.

BTW: I can talk only abstract here, what is technically happening there is beyond my horizon